File: 백준/Silver/1940. 주몽/주몽.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int m = Integer.parseInt(br.readLine());
        int[] nums = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) {
            nums[i] = Integer.parseInt(st.nextToken());
        }
    
        int left = 0;
        int right = n - 1;
        int count = 0;
    
        Arrays.sort(nums);
    
        while (left < right) {
            int sum = nums[left] + nums[right];
            if (sum < m) left++;
            else if (sum > m) right--;
            else {
                count++;
                left++;
                right--;
            }
        }
    
        System.out.println(count);   
    }
}

File: 백준/Silver/1940. 주몽/주몽.kt
import java.io.BufferedReader
import java.io.InputStreamReader
import java.util.StringTokenizer

fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val n = br.readLine().toInt()
    val m = br.readLine().toInt()
    val nums = IntArray(n)
    val st = StringTokenizer(br.readLine())
    for (i in 0 until n) {
        nums[i] = st.nextToken().toInt()
    }
    
    var left = 0
    var right = n - 1
    var count = 0
    
    nums.sort()
    
    while (left < right) {
        val sum = nums[left] + nums[right]
        when {
            sum < m -> left++
            sum > m -> right--
            else -> {
                count++
                left++
                right--
            }
        }
    }

    println(count)
}